website/content/docs/agent/telemetry.mdx

```diff
@@ -421,8 +421,8 @@ These metrics are used to monitor the health of the Consul servers.
-| `consul.rpc.query`                                  | Increments when a server receives a new blocking RPC request, indicating the rate of new blocking query calls. See consul.rpc.queries_blocking for the current number of in-flight blocking RPC calls. This metric changed in 1.7.0 to only increment on the the start of a query. The rate of queries will appear lower, but is more accurate.                                                                                                                                                                                                                                                                                                                      | queries                           | counter |
-| `consul.rpc.queries_blocking`                       | The current number of in-flight blocking queries the server is handling.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | queries                           | gauge   |
+| `consul.rpc.query`                                  | Increments when a server receives a read RPC request, indicating the rate of new read queries. See consul.rpc.queries_blocking for the current number of in-flight RPC calls. This metric changed in 1.7.0 to only increment on the the start of a query. The rate of queries will appear lower, but is more accurate.                                                                                                                                                                                                                                                                                                                      | queries                           | counter |
+| `consul.rpc.queries_blocking`                       | The current number of in-flight read queries the server is handling.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | queries                           | gauge   |
```

Comment:
Perhaps we could elaborate on this. The _value_ here is that for the most part this will be dominated by actually blocking requests at any given sample point rather than the in-flight reads that will return immediately.

So this is technically more accurate, but the distinction being made in this PR seems pretty subtle. Perhaps it would help operators interpret if we were more explicit? Maybe something like:

```suggestion
| `consul.rpc.queries_blocking`                       | The current number of in-flight read queries the server is handling. This counts all in-flight read requests whether they are blocked on updates or not, but in general will be dominated by blocked requests and can be used to monitor overall blocking query load on the server.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | queries                           | gauge   |
```

Reply:
> for the most part this will be dominated by actually blocking requests at any given sample point rather than the in-flight reads that will return immediately

I don't think that is always a fair assumption. Some times it will be true, but it's also very possible that the server is handling many non-blocking queries and the few blocking queries are a small percentage of this value.

Instead of attempting to explain the metric maybe we should fix it? I think we could only report the metric when `MinQueryIndex > 0`. I'll make that change in this PR, and restore the text for `queries_blocking`.

Reply:
Oh, actually looking again, `queries_blocking` is correct, I was mislead by the `GOTO` I think. So I'll revert the docs changes to `queries_blocking`.